lib/readify.js

```diff
@@ -31,24 +31,77 @@ const nicki = !WIN && !BROWSER && require('nicki/legacy');
 const readdir = promisify(fs.readdir, fs);
 
 /* sorting on Win and node v0.8.0 */
-const sortFiles = sort((a, b) => {
-    return a.name > b.name ? 1 : -1;
+const sortFiles = currify((attr, order, array) => {
+    switch (order) {
+    case 'desc':
+        // nothing
+        break;
+    case 'asc':
+        // nothing
+        break;
+    default:
+        order = 'asc';
+    }
+    var cmp;
+    switch (attr) {
+    case 'size':
+        cmp = (a, b) => (+a - +b);
+        break;
+    case 'date':
+        cmp = (a, b) => (a > b ? 1 : -1);
+        break;
+    case 'owner':
+        cmp = (a, b) => (a > b ? 1 : -1);
+        break;
+    case 'name':
+        cmp = (a, b) => a.localeCompare(b.attr);
+        break;
+    default:
+        attr = 'name';
+        cmp = (a, b) => a.localeCompare(b.attr);
+    }
+    return sort((a, b) => {
+        var res;
+        if (typeof a.raw !== 'undefined' && typeof b.raw !== 'undefined') {
+            res = cmp(a.raw[attr], b.raw[attr]);
+        } else {
+            res = cmp(a[attr], b[attr]);
+        }
+        if (order === 'desc') {
+            res = -res;
+        }
+        return res;
+    }, array);
 });
```

Comment:
Sort by seams very straightforward. We could fork it and write a backport with 'classic' attribute access.